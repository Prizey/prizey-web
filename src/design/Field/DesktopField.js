import React from 'react'
import { InputLabel, Input, FormHelperText } from '@material-ui/core'

export default ({ field, form, type = 'text', label, ...props }) => (
  <React.Fragment>
    <InputLabel htmlFor={field.name}>{label}</InputLabel>
    <Input {...props} {...field} type={type} id={field.name} />
    <FormHelperText id={`${field.name}-text`}>
      {form.touched[field.name] && form.errors[field.name]}
    </FormHelperText>
  </React.Fragment>
)
