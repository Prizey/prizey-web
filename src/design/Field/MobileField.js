import React from 'react'
import { Input, FormHelperText } from '@material-ui/core'

export default ({ field, form, type = 'text', ...props }) => (
  <React.Fragment>
    <Input
      {...props}
      {...field}
      type={type}
      id={field.name}
      placeholder={
        props.placeholder ? `${props.label} ${props.placeholder}` : props.label
      }
    />
    <FormHelperText id={`${field.name}-text`}>
      {form.touched[field.name] && form.errors[field.name]}
    </FormHelperText>
  </React.Fragment>
)
