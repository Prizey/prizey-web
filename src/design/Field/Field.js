import React from 'react'
import withWidth, { isWidthUp } from '@material-ui/core/withWidth'
import { FormControl } from '@material-ui/core'

import MobileField from './MobileField'
import DesktopField from './DesktopField'

export const InputField = ({ width, field, form, ...props }) => {
  const UseField = isWidthUp('sm', width) ? DesktopField : MobileField
  return (
    <FormControl
      component="fieldset"
      error={form.touched[field.name] && !!form.errors[field.name]}
      id={`${field.name}-fieldset`}
      fullWidth
    >
      <UseField {...props} field={field} form={form} />
    </FormControl>
  )
}

export default withWidth()(InputField)
