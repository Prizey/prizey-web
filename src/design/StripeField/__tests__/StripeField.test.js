import React from 'react'
import renderer from 'react-test-renderer'

import Field, {
  reducer,
  handleFocus,
  handleBlur,
  handleChange,
} from '../StripeField'

it('renders correctly the desktop field', () => {
  const tree = renderer
    .create(<Field label="field" component={props => <input {...props} />} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('test the reducer hook', () => {
  const initialState = {
    empty: true,
    error: null,
    focused: false,
  }
  let state = {}

  state = reducer(initialState, {})
  expect(state).toEqual(initialState)

  state = reducer(state, { type: 'focus' })
  expect(state.focused).toBeTruthy()

  state = reducer(state, { type: 'blur' })
  expect(state.focused).toBeFalsy()

  state = reducer(state, {
    payload: { empty: false, error: true },
    type: 'change',
  })
  expect(state.empty).toBeFalsy()
  expect(state.error).toBeTruthy()
})

it('handleFocus will set the focus', () => {
  const dispatch = jest.fn()
  const action = { type: 'focus' }

  handleFocus(dispatch)()
  expect(dispatch).toHaveBeenCalledWith(action)
})

it('handleBlur will set the focus', () => {
  const dispatch = jest.fn()
  const action = { type: 'blur' }

  handleBlur(dispatch)()
  expect(dispatch).toHaveBeenCalledWith(action)
})

it('handleChange will set the focus', () => {
  const dispatch = jest.fn()
  const action = {
    payload: { empty: false, error: 'the error' },
    type: 'change',
  }

  handleChange(dispatch)(action.payload)
  expect(dispatch).toHaveBeenCalledWith(action)
})
