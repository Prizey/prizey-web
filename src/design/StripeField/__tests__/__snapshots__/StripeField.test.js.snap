// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the desktop field 1`] = `
<fieldset
  className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
>
  <label
    className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
    data-shrink={false}
  >
    field
  </label>
  <div
    className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
    onClick={[Function]}
  >
    <input
      className="Component-root-53"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "base": Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "14px",
          },
          "invalid": Object {
            "color": "#f44336",
          },
        }
      }
    />
  </div>
  <p
    className="MuiFormHelperText-root-54"
  />
</fieldset>
`;
